### PR TITLE
XOR-629 [FEAT] Add support for ID generator to Text Input field

### PR DESCRIPTION
**Pull request by NikitaJadhavXoriant**
### Product areas affected

Widgets -> Add Form -> Select Text Input Field -> Check / Uncheck 'Have this form field act as an ID generator 

### What does this PR do?

Implemented code to convert Text input form field into ID generator.

### JIRA ticket

[JIRA](https://weboo.atlassian.net/browse/XOR-629)

### Result

https://user-images.githubusercontent.com/108272606/198697877-36c682e0-8490-4349-ac78-f96c2702f63a.mp4

### Checklist

None

### Deployment instructions

None

### Author concerns

None

### CODE REVIEW

**Review by amolsuryawanshi47 — DISMISSED** [review]
Changes Looks ok to me

File: js/components/input.js [hunk 1]
```diff
@@ -19,5 +27,33 @@ Fliplet.FormBuilder.field('input', {
...
+        this.getNewGuid();
+      }
+    }
+  },
+  created: function() {
+    if (!this.valueIsFromProgress) {
+      this.getNewGuid();
+    }
+
+    Fliplet.FormBuilder.on('reset', this.onReset, this.$parent.id);
```
Comment (tonytlwu) [inline js/components/input.js:54]:
```suggestion
    Fliplet.FormBuilder.on('reset', this.onReset);
```

File: js/components/input.js [hunk 2]
```diff
@@ -19,5 +27,33 @@ Fliplet.FormBuilder.field('input', {
...
+  },
+  destroyed: function() {
+    Fliplet.FormBuilder.off('reset', this.onReset, this.$parent.id);
```
Comment (tonytlwu) [inline js/components/input.js:57]:
```suggestion
    Fliplet.FormBuilder.off('reset', this.onReset);
```

File: js/libs/core.js [hunk 3]
```diff
@@ -14,14 +14,14 @@ Fliplet.FormBuilder = (function() {
   }
 
   return {
-    on: function(eventName, fn) {
-      eventHub.$on(eventName, fn);
+    on: function(eventName, fn, id) {
```
Comment (tonytlwu) [inline js/libs/core.js:17]: Same for `off` and `emit`.

```suggestion
    on: function(eventName, fn) {
```

File: js/libs/core.js [hunk 4]
```diff
@@ -14,14 +14,14 @@ Fliplet.FormBuilder = (function() {
...
+      eventHub.$on(eventName + (id || ''), fn);
```
Comment (tonytlwu) [inline js/libs/core.js:18]: Same for `off` and `emit`.

```suggestion
      eventHub.$on(eventName, fn);
```

**Review by tonytlwu — CHANGES_REQUESTED** [review]

File: js/libs/core.js (hunk 4 above)
Comment (NikitaJadhavXoriant) [reply js/libs/core.js]: Done

File: js/libs/core.js (hunk 3 above)
Comment (NikitaJadhavXoriant) [reply js/libs/core.js]: Done

File: js/components/input.js (hunk 2 above)
Comment (NikitaJadhavXoriant) [reply js/components/input.js]: Done

File: js/components/input.js (hunk 1 above)
Comment (NikitaJadhavXoriant) [reply js/components/input.js]: Done

**Review by amolsuryawanshi47 — APPROVED** [review]
Changes looks ok to me

**Review by tonytlwu — APPROVED** [review]

**Pull request merged**